### ISSUE TITLE
TypeORM and TS.ed after routes init not started

### ISSUE THREAD

**Issue by OskarLebuda** [Romakita/ts-express-decorators#459]
## Informations

Type | Version
---|---
Bug/ | 4.x

****

## Description
Hello.
I have been trying to integrate TS.ed with TypeORM for a long time. I found a potential error that does not allow simultaneous integration and using `componentsScan`. Ff I add the `typeorm` and` componentsScan` options simultaneously when the server is started, the `$ afterRoutesInit` hook will not be executed in my service.

EXAMPLE:

``` typescript
@ServerSettings({
    rootDir,
    mount: {
        '/v1': `${rootDir}/controllers/**/**.controller.{ts,js}`
    },
    componentsScan: [
        `${rootDir}/middlewares/**/**.{ts,js}`,
        `${rootDir}/services/**/**.{ts,js}`,
    ],
    typeorm: [
        {
            name: "default",
            type: "postgres",
            host: "0.0.0.0",
            port: 5432,
            username: "postgres",
            password: "postgres",
            database: "postgres",
            synchronize: true,
            logging: false,
            entities: [
                `${__dirname}/entity/*{.ts,.js}`
            ],
            migrations: [
                `${__dirname}/migrations/*{.ts,.js}`
            ],
            subscribers: [
                `${__dirname}/subscriber/*{.ts,.js}`
            ]
        }
    ],
    httpPort: process.env.PORT || 3000,
    httpsPort: false,
    acceptMimes: ['application/json', ''],
    swagger: {
        path: '/doc'
    },
})
```

```  typescript
@Service()
export class UserService implements AfterRoutesInit {

    private connection: Connection;

    constructor(
        private typeORMService: TypeORMService
    ) {
    }

    $afterRoutesInit() {
        this.connection = this.typeORMService.get();
        $log.warn('Connected to the Users');
    }
}
```

In this case the `$ afterRoutesInit` hook will not be executed. 

Can someone help me? Is there any solution that can be circumvented by this error?

## Acceptance criterias

- [x] Example (log, usecase, etc...)
- [x] Ts.ED version (bug/question)

**Comment by Romakita** [comment]
Ok I'll check in the example here ASAP https://github.com/Romakita/example-ts-express-decorator/blob/4.0.0/example-typeorm

**Comment by OskarLebuda** [comment]
@Romakita in your example there's no information about `componentsScan`

**Comment by OskarLebuda** [comment]
@Romakita any results? I have to fix it asap

**Comment by Romakita** [comment]
@OskarLebuda I'd just tested and push working example with componentScan.
I the $afterRoutesInit isn't triggered correctly it means the componentScan isn't correctly configured.

Are sure the rootDir is good ? Because for me rootDir should have __dirname as value.
```typescript
    mount: {
        '/v1': `${rootDir}/controllers/**/**.controller.{ts,js}`
    },
    componentsScan: [
        `${rootDir}/middlewares/**/**.{ts,js}`,
        `${rootDir}/services/**/**.{ts,js}`,
    ],
```
Can you test the example on example-ts-decorator and compare your project with?
See you,
Romain

**Comment by Romakita** [comment]
You can add `debug: true` on ServerSettings decorator to display Services/Controllers built by the Server.

**Comment by OskarLebuda** [comment]
@Romakita 

Ok I see. The problem is in the middleware. I Inject UserService to my auth service which is inject to my auth middleware like:

`auth.middleware.ts`
``` typescript

@OverrideMiddleware(AuthenticatedMiddleware)
export class AuthMiddleware implements IMiddleware  {

    constructor(private authService: AuthService) {
    }

    async use(@EndpointInfo() endpoint: EndpointMetadata, @Req() request: any) {
        // auth logic
    }

}
```
`auth.service.ts`
```
@Service()
export class AuthService {

    constructor(
        @Inject() userService: UserService,
    ) {
    }
```

`user.service.ts`
```

@Service()
export class UserService implements OnInit, BeforeRoutesInit, OnRoutesInit, AfterRoutesInit, OnServerReady {

    private connection: Connection;

    constructor(
        @Inject() private typeORMService: TypeORMService
    ) {
    }

    $onInit(): Promise<any> | void {
        $log.warn('All services is ready');
    }

    $beforeRoutesInit(): Promise<any> | void {
        $log.warn('Controllers and routes isn\'t mounted');
    }

    $onRoutesInit(components: IComponentScanned[]): Promise<any> | void {
        $log.warn('Controllers and routes are being built');

    }

    $afterRoutesInit(): Promise<any> | void {
        $log.warn('Controllers and routes are built');
    }

    $onServerReady(): Promise<any> | void {
        $log.warn('Server is ready and listen the port');
    }
```

In this case hooks will not be executed.

**Comment by OskarLebuda** [comment]
@Romakita any results? I have to fix it asap

**Comment by Romakita** [comment]
@OskarLebuda Are you sure the UserService is correctly loaded by the componentScan ?
Because Service Lifecycle works fine for me and the example with TypeORM works fine with the componentScan configuration (and by default, Ts.ED have a componentScan configured).

To know if your service is correctly loaded and built, add `debug: true` on your server, then send me de log displayed by the server.

```typescript
@ServerSettings({
   debug: true, // Add this line
}) 
```

Note: `@Inject` is not necessary to Inject service in another service when the Symbol is a class.

See you,
Romain

**Comment by Romakita** [comment]
I've added a new branch aligned with you convention naming and your configuration. This example works perfectly. No fix is required, but I'm sure your problem come from the configuration on ServerSettings ;)

https://github.com/Romakita/example-ts-express-decorator/tree/oskarlebuda/example-typeorm

See you
Romain

**Comment by OskarLebuda** [comment]
@Romakita  Actually it's not working.. Look at the log from console. The hooks are not executed. I just change the url to the database. It's 100% correct. Server starts, but after create come request I have an error. 

``` typescript
{
    host: "0.0.0.0",
    port: 5432,
    username: "postgres",
    password: "postgres",
    database: "postgres",
}
```

``` bash
[2018-09-25T19:30:00.869] [ERROR] [TSED] - {
  "error": {
    "status": 500,
    "message": "Cannot read property 'manager' of undefined",
    "stack": "TypeError: Cannot read property 'manager' of undefined\n    at UsersService.<anonymous> (/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/services/users.service.ts:47:35)\n    at Generator.next (<anonymous>)\n    at /Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/services/users.service.ts:16:71\n    at new Promise (<anonymous>)\nat __awaiter (/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/services/users.service.ts:12:12)\n    at UsersService.create (/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/services/users.service.ts:46:16)\n    at UsersCtrl.create (/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/controllers/users/users.ctrl.ts:31:34)\n    at HandlerBuilder.<anonymous> (/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/node_modules/@tsed/common/src/mvc/class/HandlerBuilder.ts:117:67)\n    at Generator.next (<anonymous>)\n    at /Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/node_modules/tslib/tslib.js:107:75"
  },
  "reqId": "1",
  "method": "POST",
  "url": "/v1/users",
  "duration": 36,
  "headers": {
    "content-type": "application/json",
    "authorization": "Bearer {{authToken}}",
    "content-length": "69",
    "host": "0.0.0.0:8080",
    "connection": "Keep-Alive",
    "user-agent": "Apache-HttpClient/4.5.5 (Java/1.8.0_152-release)",
    "accept-encoding": "gzip,deflate"
  },
  "body": {
    "firstName": "Oskar",
    "lastName": "Lebuda",
    "age": 10
  },
  "query": {},
  "params": {}
}
```

Server start info

``` bash
> test-ts-express-decorator@1.0.0 start /Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm
> ts-node src/index.ts

[2018-09-25T19:25:23.665] [DEBUG] [TSED] - Initialize settings
[2018-09-25T19:25:23.668] [INFO ] [TSED] - settings.rootDir => /Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src
[2018-09-25T19:25:23.668] [INFO ] [TSED] - settings.env => development
[2018-09-25T19:25:23.668] [INFO ] [TSED] - settings.httpPort => 8080
[2018-09-25T19:25:23.668] [INFO ] [TSED] - settings.httpsPort => 8000
[2018-09-25T19:25:23.668] [INFO ] [TSED] - settings.version => 1.0.0
[2018-09-25T19:25:23.668] [INFO ] [TSED] - settings.uploadDir => /Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/uploads
[2018-09-25T19:25:23.668] [INFO ] [TSED] - settings.logger => { requestFields: undefined,
  debug: true,
  logRequest: true,
  jsonIndentation: 2 }
[2018-09-25T19:25:23.670] [INFO ] [TSED] - settings.scope => singleton
[2018-09-25T19:25:23.670] [INFO ] [TSED] - settings.errors => { headerName: 'errors' }
[2018-09-25T19:25:23.674] [INFO ] [TSED] - settings.mount => { '/v1': '/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/controllers/**/*.ctrl.{ts,js}' }
[2018-09-25T19:25:23.674] [INFO ] [TSED] - settings.exclude => [ '**/*.spec.ts', '**/*.spec.js' ]
[2018-09-25T19:25:23.675] [INFO ] [TSED] - settings.componentsScan => [ '/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/middlewares/**/*.middleware.{ts,js}',
  '/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/services/**/*.service.{ts,js}' ]
[2018-09-25T19:25:23.675] [INFO ] [TSED] - settings.acceptMimes => [ 'application/json' ]
[2018-09-25T19:25:23.675] [INFO ] [TSED] - settings.passport => {}
[2018-09-25T19:25:23.675] [INFO ] [TSED] - settings.typeorm => [ { name: 'default',
    type: 'postgres',
    host: '0.0.0.0',
    port: 5432,
    username: 'postgres',
    password: 'postgres',
    database: 'postgres',
    synchronize: true,
    logging: false,
    entities:
     [ '/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/entity/*{.ts,.js}' ],
    migrations:
     [ '/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/migrations/*{.ts,.js}' ],
    subscribers:
     [ '/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/subscriber/*{.ts,.js}' ] } ]
[2018-09-25T19:25:23.675] [INFO ] [TSED] - settings.swagger => { path: '/api-docs' }
[2018-09-25T19:25:23.675] [INFO ] [TSED] - Build services
[2018-09-25T19:25:23.677] [DEBUG] [TSED] - InjectorService loaded
[2018-09-25T19:25:23.677] [DEBUG] [TSED] - ServerSettingsService loaded
[2018-09-25T19:25:23.677] [DEBUG] [TSED] - ExpressApplication loaded
[2018-09-25T19:25:23.677] [DEBUG] [TSED] - HttpServer loaded
[2018-09-25T19:25:23.677] [DEBUG] [TSED] - HttpsServer loaded
[2018-09-25T19:25:23.677] [DEBUG] [TSED] - JsonSchemesService built
[2018-09-25T19:25:23.677] [DEBUG] [TSED] - ConverterService built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - ArrayConverter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - DateConverter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - MapConverter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - PrimitiveConverter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - SetConverter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - SymbolConverter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - ParseService built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - BodyParamsFilter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - CookiesFilter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - HeaderParamsFilter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - LocalsFilter built
[2018-09-25T19:25:23.678] [DEBUG] [TSED] - PathParamsFilter built
[2018-09-25T19:25:23.679] [DEBUG] [TSED] - QueryParamsFilter built
[2018-09-25T19:25:23.679] [DEBUG] [TSED] - SessionFilter built
[2018-09-25T19:25:23.679] [DEBUG] [TSED] - FilterService built
[2018-09-25T19:25:23.679] [DEBUG] [TSED] - ValidationService built
[2018-09-25T19:25:23.679] [DEBUG] [TSED] - SendResponseMiddleware built
[2018-09-25T19:25:23.679] [DEBUG] [TSED] - GlobalAcceptMimesMiddleware built
[2018-09-25T19:25:23.679] [DEBUG] [TSED] - GlobalErrorHandlerMiddleware built
[2018-09-25T19:25:23.680] [WARN ] [TSED] - ===> UsersService { typeORMService: TypeORMService { _instances: Map {} } }
[2018-09-25T19:25:23.680] [WARN ] [TSED] - AuthMiddleware built
[2018-09-25T19:25:23.680] [DEBUG] [TSED] - AuthenticatedMiddleware built from class AuthMiddleware
[2018-09-25T19:25:23.680] [DEBUG] [TSED] - ResponseViewMiddleware built
[2018-09-25T19:25:23.680] [DEBUG] [TSED] - RouteService built
[2018-09-25T19:25:23.683] [DEBUG] [TSED] - ControllerService built
[2018-09-25T19:25:23.683] [DEBUG] [TSED] - MiddlewareService built
[2018-09-25T19:25:23.684] [DEBUG] [TSED] - LogIncomingRequestMiddleware built
[2018-09-25T19:25:23.684] [DEBUG] [TSED] - SwaggerService built
[2018-09-25T19:25:23.684] [DEBUG] [TSED] - SwaggerModule built
[2018-09-25T19:25:23.684] [DEBUG] [TSED] - TypeORMModule built
[2018-09-25T19:25:23.684] [DEBUG] [TSED] - RestCtrl built
[2018-09-25T19:25:23.684] [DEBUG] [TSED] - UsersCtrl built
[2018-09-25T19:25:23.685] [DEBUG] [TSED] - Call hook $onInit
[2018-09-25T19:25:23.685] [INFO ] [TSED] - Create connection with typeorm to database: default
[2018-09-25T19:25:23.685] [DEBUG] [TSED] - options: {"name":"default","type":"postgres","host":"0.0.0.0","port":5432,"username":"postgres","password":"postgres","database":"postgres","synchronize":true,"logging":false,"entities":["/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/entity/*{.ts,.js}"],"migrations":["/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/migrations/*{.ts,.js}"],"subscribers":["/Users/oskarlebuda/Downloads/example-ts-express-decorator-oskarlebuda/example-typeorm/src/subscriber/*{.ts,.js}"]}
[2018-09-25T19:25:23.710] [DEBUG] [TSED] - Call hook $onInit  promises built
[2018-09-25T19:25:23.766] [INFO ] [TSED] - Connected with typeorm to database: default
[2018-09-25T19:25:23.767] [DEBUG] [TSED] - Settings and injector loaded
[2018-09-25T19:25:23.767] [DEBUG] [TSED] - Mount middlewares
[2018-09-25T19:25:23.768] [DEBUG] [TSED] - Call hook $onMountingMiddlewares
[2018-09-25T19:25:23.814] [DEBUG] [TSED] - Call hook $beforeRoutesInit
[2018-09-25T19:25:23.815] [DEBUG] [TSED] - Call hook $onRoutesInit
[2018-09-25T19:25:23.815] [INFO ] [TSED] - Map controllers
[2018-09-25T19:25:23.815] [DEBUG] [TSED] - Call hook $afterRoutesInit
[2018-09-25T19:25:23.815] [INFO ] [TSED] - Routes mounted :
[2018-09-25T19:25:23.816] [INFO ] [TSED] -
┌────────────────┬────────────┬──────────────────────┐
│ Method         │ Endpoint   │ Class method         │
│────────────────│────────────│──────────────────────│
│ GET            │ /v1/rest/  │ RestCtrl.getRoutes() │
│────────────────│────────────│──────────────────────│
│ POST           │ /v1/users/ │ UsersCtrl.create()   │
│────────────────│────────────│──────────────────────│
│ GET            │ /v1/users/ │ UsersCtrl.getList()  │
└────────────────┴────────────┴──────────────────────┘
[2018-09-25T19:25:23.859] [DEBUG] [TSED] - Start server on http://0.0.0.0:8080
[2018-09-25T19:25:23.859] [DEBUG] [TSED] - Start server on https://0.0.0.0:8000
[2018-09-25T19:25:23.861] [INFO ] [TSED] - HTTP Server listen on http://0.0.0.0:8080
[2018-09-25T19:25:23.861] [INFO ] [TSED] - HTTP Server listen on https://0.0.0.0:8000
[2018-09-25T19:25:23.862] [DEBUG] [TSED] - Call hook $onServerReady
[2018-09-25T19:25:23.862] [INFO ] [TSED] - [default] Swagger JSON is available on http://0.0.0.0:8080/api-docs/swagger.json
[2018-09-25T19:25:23.862] [INFO ] [TSED] - [default] Swagger UI is available on http://0.0.0.0:8080/api-docs/
[2018-09-25T19:25:23.862] [INFO ] [TSED] - Started in 239 ms
```

**Comment by OskarLebuda** [comment]
I just add that everything is on the Mac OS X environment (mojave)

**Comment by Romakita** [comment]
Here my logs:

```sh

[2018-09-26T07:47:46.521] [DEBUG] [TSED] - Initialize settings
[2018-09-26T07:47:46.524] [INFO ] [TSED] - settings.rootDir => /Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src
[2018-09-26T07:47:46.525] [INFO ] [TSED] - settings.env => development
[2018-09-26T07:47:46.525] [INFO ] [TSED] - settings.httpPort => 8080
[2018-09-26T07:47:46.525] [INFO ] [TSED] - settings.httpsPort => 8000
[2018-09-26T07:47:46.525] [INFO ] [TSED] - settings.version => 1.0.0
[2018-09-26T07:47:46.525] [INFO ] [TSED] - settings.uploadDir => /Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/uploads
[2018-09-26T07:47:46.525] [INFO ] [TSED] - settings.logger => { requestFields: undefined,
  debug: true,
  logRequest: true,
  jsonIndentation: 2 }
[2018-09-26T07:47:46.526] [INFO ] [TSED] - settings.scope => singleton
[2018-09-26T07:47:46.526] [INFO ] [TSED] - settings.errors => { headerName: 'errors' }
[2018-09-26T07:47:46.527] [INFO ] [TSED] - settings.mount => { '/v1': '/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/controllers/**/*.ctrl.{ts,js}' }
[2018-09-26T07:47:46.527] [INFO ] [TSED] - settings.exclude => [ '**/*.spec.ts', '**/*.spec.js' ]
[2018-09-26T07:47:46.527] [INFO ] [TSED] - settings.componentsScan => [ '/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/middlewares/**/*.middleware.{ts,js}',
  '/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/services/**/*.service.{ts,js}' ]
[2018-09-26T07:47:46.527] [INFO ] [TSED] - settings.acceptMimes => [ 'application/json' ]
[2018-09-26T07:47:46.527] [INFO ] [TSED] - settings.passport => {}
[2018-09-26T07:47:46.528] [INFO ] [TSED] - settings.typeorm => [ { name: 'default',
    type: 'postgres',
    host: 'localhost',
    port: 5432,
    username: 'postgres',
    password: '',
    database: 'testdb',
    synchronize: true,
    logging: false,
    entities: 
     [ '/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/entity/*{.ts,.js}' ],
    migrations: 
     [ '/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/migrations/*{.ts,.js}' ],
    subscribers: 
     [ '/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/subscriber/*{.ts,.js}' ] } ]
[2018-09-26T07:47:46.528] [INFO ] [TSED] - settings.swagger => { path: '/api-docs' }
[2018-09-26T07:47:46.528] [INFO ] [TSED] - Build services
[2018-09-26T07:47:46.530] [DEBUG] [TSED] - InjectorService loaded
[2018-09-26T07:47:46.530] [DEBUG] [TSED] - ServerSettingsService loaded
[2018-09-26T07:47:46.530] [DEBUG] [TSED] - ExpressApplication loaded
[2018-09-26T07:47:46.530] [DEBUG] [TSED] - HttpServer loaded
[2018-09-26T07:47:46.530] [DEBUG] [TSED] - HttpsServer loaded
[2018-09-26T07:47:46.531] [DEBUG] [TSED] - JsonSchemesService built 
[2018-09-26T07:47:46.531] [DEBUG] [TSED] - ConverterService built 
[2018-09-26T07:47:46.532] [DEBUG] [TSED] - ArrayConverter built 
[2018-09-26T07:47:46.532] [DEBUG] [TSED] - DateConverter built 
[2018-09-26T07:47:46.532] [DEBUG] [TSED] - MapConverter built 
[2018-09-26T07:47:46.532] [DEBUG] [TSED] - PrimitiveConverter built 
[2018-09-26T07:47:46.533] [DEBUG] [TSED] - SetConverter built 
[2018-09-26T07:47:46.534] [DEBUG] [TSED] - SymbolConverter built 
[2018-09-26T07:47:46.534] [DEBUG] [TSED] - ParseService built 
[2018-09-26T07:47:46.534] [DEBUG] [TSED] - BodyParamsFilter built 
[2018-09-26T07:47:46.534] [DEBUG] [TSED] - CookiesFilter built 
[2018-09-26T07:47:46.534] [DEBUG] [TSED] - HeaderParamsFilter built 
[2018-09-26T07:47:46.535] [DEBUG] [TSED] - LocalsFilter built 
[2018-09-26T07:47:46.535] [DEBUG] [TSED] - PathParamsFilter built 
[2018-09-26T07:47:46.535] [DEBUG] [TSED] - QueryParamsFilter built 
[2018-09-26T07:47:46.535] [DEBUG] [TSED] - SessionFilter built 
[2018-09-26T07:47:46.536] [DEBUG] [TSED] - FilterService built 
[2018-09-26T07:47:46.536] [DEBUG] [TSED] - ValidationService built 
[2018-09-26T07:47:46.536] [DEBUG] [TSED] - SendResponseMiddleware built 
[2018-09-26T07:47:46.536] [DEBUG] [TSED] - GlobalAcceptMimesMiddleware built 
[2018-09-26T07:47:46.536] [DEBUG] [TSED] - GlobalErrorHandlerMiddleware built 
[2018-09-26T07:47:46.537] [WARN ] [TSED] - ===> UsersService { typeORMService: TypeORMService { _instances: Map {} } }
[2018-09-26T07:47:46.537] [WARN ] [TSED] - AuthMiddleware built
[2018-09-26T07:47:46.537] [DEBUG] [TSED] - AuthenticatedMiddleware built from class AuthMiddleware
[2018-09-26T07:47:46.537] [DEBUG] [TSED] - ResponseViewMiddleware built 
[2018-09-26T07:47:46.538] [DEBUG] [TSED] - RouteService built 
[2018-09-26T07:47:46.542] [DEBUG] [TSED] - ControllerService built 
[2018-09-26T07:47:46.542] [DEBUG] [TSED] - MiddlewareService built 
[2018-09-26T07:47:46.542] [DEBUG] [TSED] - LogIncomingRequestMiddleware built 
[2018-09-26T07:47:46.542] [DEBUG] [TSED] - SwaggerService built 
[2018-09-26T07:47:46.543] [DEBUG] [TSED] - SwaggerModule built 
[2018-09-26T07:47:46.543] [DEBUG] [TSED] - TypeORMModule built 
[2018-09-26T07:47:46.543] [DEBUG] [TSED] - RestCtrl built 
[2018-09-26T07:47:46.543] [DEBUG] [TSED] - RestCtrl built 
[2018-09-26T07:47:46.544] [DEBUG] [TSED] - UsersCtrl built 
[2018-09-26T07:47:46.544] [DEBUG] [TSED] - UsersCtrl built 
[2018-09-26T07:47:46.544] [WARN ] [TSED] - ===> UsersService { typeORMService: TypeORMService { _instances: Map {} } }
[2018-09-26T07:47:46.544] [DEBUG] [TSED] - AuthService built 
[2018-09-26T07:47:46.544] [DEBUG] [TSED] - UsersService built 
[2018-09-26T07:47:46.545] [DEBUG] [TSED] - Call hook $onInit 
[2018-09-26T07:47:46.545] [INFO ] [TSED] - Create connection with typeorm to database: default
[2018-09-26T07:47:46.545] [DEBUG] [TSED] - options: {"name":"default","type":"postgres","host":"localhost","port":5432,"username":"postgres","password":"","database":"testdb","synchronize":true,"logging":false,"entities":["/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/entity/*{.ts,.js}"],"migrations":["/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/migrations/*{.ts,.js}"],"subscribers":["/Users/romain.lenzotti/workspace/ts-projects/example-ts-express-decorator/example-typeorm/src/subscriber/*{.ts,.js}"]}
[2018-09-26T07:47:46.580] [WARN ] [TSED] - All services is ready
[2018-09-26T07:47:46.581] [DEBUG] [TSED] - Call hook $onInit  promises built
[2018-09-26T07:47:46.886] [INFO ] [TSED] - Connected with typeorm to database: default
[2018-09-26T07:47:46.886] [DEBUG] [TSED] - Settings and injector loaded
[2018-09-26T07:47:46.887] [DEBUG] [TSED] - Mount middlewares
[2018-09-26T07:47:46.887] [DEBUG] [TSED] - Call hook $onMountingMiddlewares
[2018-09-26T07:47:46.932] [DEBUG] [TSED] - Call hook $beforeRoutesInit 
[2018-09-26T07:47:46.932] [WARN ] [TSED] - Controllers and routes isn't mounted
[2018-09-26T07:47:46.933] [DEBUG] [TSED] - Call hook $onRoutesInit 
[2018-09-26T07:47:46.933] [INFO ] [TSED] - Map controllers
[2018-09-26T07:47:46.934] [WARN ] [TSED] - Controllers and routes are being built
[2018-09-26T07:47:46.934] [DEBUG] [TSED] - Call hook $afterRoutesInit 
[2018-09-26T07:47:46.934] [INFO ] [TSED] - Routes mounted :
[2018-09-26T07:47:46.936] [INFO ] [TSED] - 
┌────────────────┬────────────┬──────────────────────┐
│ Method         │ Endpoint   │ Class method         │
│────────────────│────────────│──────────────────────│
│ GET            │ /v1/rest/  │ RestCtrl.getRoutes() │
│────────────────│────────────│──────────────────────│
│ GET            │ /v1/rest/  │ RestCtrl.getRoutes() │
│────────────────│────────────│──────────────────────│
│ POST           │ /v1/users/ │ UsersCtrl.create()   │
│────────────────│────────────│──────────────────────│
│ GET            │ /v1/users/ │ UsersCtrl.getList()  │
│────────────────│────────────│──────────────────────│
│ POST           │ /v1/users/ │ UsersCtrl.create()   │
│────────────────│────────────│──────────────────────│
│ GET            │ /v1/users/ │ UsersCtrl.getList()  │
└────────────────┴────────────┴──────────────────────┘
[2018-09-26T07:47:46.942] [WARN ] [TSED] - Controllers and routes are built
this.connection => true
[2018-09-26T07:47:46.943] [DEBUG] [TSED] - Start server on http://0.0.0.0:8080
[2018-09-26T07:47:46.944] [DEBUG] [TSED] - Start server on https://0.0.0.0:8000
[2018-09-26T07:47:46.947] [INFO ] [TSED] - HTTP Server listen on http://0.0.0.0:8080
[2018-09-26T07:47:46.947] [INFO ] [TSED] - HTTP Server listen on https://0.0.0.0:8000
[2018-09-26T07:47:46.948] [DEBUG] [TSED] - Call hook $onServerReady 
[2018-09-26T07:47:46.948] [INFO ] [TSED] - [default] Swagger JSON is available on http://0.0.0.0:8080/api-docs/swagger.json
[2018-09-26T07:47:46.948] [INFO ] [TSED] - [default] Swagger UI is available on http://0.0.0.0:8080/api-docs/
[2018-09-26T07:47:46.948] [WARN ] [TSED] - Server is ready and listen the port
[2018-09-26T07:47:46.949] [INFO ] [TSED] - Started in 446 ms
```

**Comment by Romakita** [comment]
Ok I found a problem. It come from ts-node and tsc in the example. let me check the problem.

**Comment by OskarLebuda** [comment]
Great. Let me know if you find something 👍 🎉

**Comment by Romakita** [comment]
@OskarLebuda Great news it's fixed with v4.30.9.
I've updated the example ;)

See you
Romain

**Comment by OskarLebuda** [comment]
It works! Thanks for your help! We can close this task
